### PR TITLE
OPSEXP-1620 Update ACS versions via updatecli

### PR DESCRIPTION
**Pull request by gionn**
https://www.updatecli.io/docs/prologue/introduction/

OPSEXP-1620

update group_vars/all.yml versions with:

```
updatecli apply --config scripts/updatecli/updatecli_config.yml --values scripts/updatecli/updatecli_acs73.yml
```

- support the versions specified in molecule tests

### CODE REVIEW

File: scripts/updatecli/updatecli_acs72.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  version: "2.9"
+
+aos:
+  version: "1.4"
+
+apiExplorer:
+  version: "7.2"
+
+dsync:
+  version: "3.6"
```
Comment (alxgomz) [inline scripts/updatecli/updatecli_acs72.yml:14]: 7.2 also support 3.7

File: scripts/updatecli/updatecli_acs72.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+acs:
+  version: "7.2"
+
+adw:
+  version: "2.9"
```
Comment (alxgomz) [inline scripts/updatecli/updatecli_acs72.yml:5]: also supports 3.0

File: scripts/updatecli/updatecli_acs72.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  version: "2.0"
+
+searchEnterprise:
+  version: "3.1"
+
+transform:
+  version: "2.5"
+
+trouter:
+  version: "1.5"
```
Comment (alxgomz) [inline scripts/updatecli/updatecli_acs72.yml:29]: as trouter & transform versions have been bumped respectively to `2.x` & `3.x` I would use

```suggestion
transform:
  version: "2"

trouter:
  version: "1"
```

File: scripts/updatecli/updatecli_config.yml [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+        kind: regex
+        pattern: '{{ .searchEnterprise.version }}.(\d+){{ .version_pattern }}'
+  transform:
+    name: Transform Core {{ .transform.version }}.x
+    kind: gittag
+    scmid: transformRepo
+    spec:
+      versionFilter:
+        kind: regex
+        pattern: '{{ .transform.version }}.(\d+){{ .version_pattern }}'
```
Comment (alxgomz) [inline scripts/updatecli/updatecli_config.yml:137]:
```suggestion
        pattern: '{{ .transform.version }}(.\d+)+{{ .version_pattern }}'
```

File: scripts/updatecli/updatecli_config.yml [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+  trouter:
+    name: ATS Transform Service {{ .trouter.version }}.x
+    kind: gittag
+    scmid: trouterRepo
+    spec:
+      versionFilter:
+        kind: regex
+        pattern: '{{ .trouter.version }}.(\d+){{ .version_pattern }}'
```
Comment (alxgomz) [inline scripts/updatecli/updatecli_config.yml:145]:
```suggestion
        pattern: '{{ .trouter.version }}(.\d+)+{{ .version_pattern }}'
```

File: scripts/updatecli/updatecli_config.yml [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+
+targets:
+  acs:
+    name: Bump acs
+    kind: yaml
+    sourceid: acs
+    spec:
+      key: acs.version
+      file: group_vars/all.yml
```
Comment (alxgomz) [inline scripts/updatecli/updatecli_config.yml:155]: Can't we use a variable from the yaml files to set the targets? (so we can support updating extra-vars files?)

**Review by alxgomz — COMMENTED** [review]
I'm not sure our versioning scheme is "stable enough" across component no we'll still need to be extra cautious when doing this task

**Comment by alxgomz** [comment]
Not sure tbh... We should clarify that with Andy first (he's on PTO for now)

**Review by alxgomz — COMMENTED** [review]
Not sure tbh... We should clarify that with Andy first (he's on PTO for now)

**Comment by gionn** [comment]
> [ ] support the versions specified in molecule tests

doesn't needs any special handling because we just need another run of updatecli pointing that file as a target:

```
target_file: roles/repository/molecule/default/host_vars/repository-instance.yml
```

and it will update only the already existing keys it can find.

File: scripts/updatecli/updatecli_acs73.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  version: "3.2"
+
+transform:
+  version: "3"
+
+trouter:
+  version: "2"
+
+target_file: group_vars/all.yml
+version_pattern: '(-(A|M)\d*)?$'
```
Comment (alxgomz) [inline scripts/updatecli/updatecli_acs73.yml:32]:
```suggestion
version_pattern: '(-M\d*)?$'
```

**Review by alxgomz — APPROVED** [review]

**Pull request merged**